### PR TITLE
5348 cache access token

### PR DESCRIPTION
**Pull request by nr-kkenney**
## Summary
This PR introduces a change to start caching the access token in getAccessToken. Additionally, getting the access token is now almost entirely encapsulated in vendor-request.js.

Now, most of the scripts don't have to even worry about access tokens -- getting and caching the token is handled in one spot. This is reflected with the changes to vendorRequest method signature, and removal of access token code usage.

We push the generation of access tokens deeper into the code, closer to where we actually need to use it. Rather than our high level scripts caring about getting & managing an access token, we leave that concern in our lower level vendor api code.

Now, every time vendorRequest is called, we call getAccessToken internally. getAccessToken will automatically generate & store valid access tokens with the caching change -- an initial request will generate a valid token, then subsequent requests will use the cached token. once the token expires, we will go out and get another one and repeat the cycle.

important to note -- the expiration for the token is set to 4 minutes in an attempt to give us a bit of leeway for using a token and not having it expire while we use it or attempt to use it.

## The Fix
How does this address the auth issues we ran into? 

Code that calls getAccessToken no longer needs to worry about the 5 minute expiration, since the internal caching is 4 minutes & retrieves a new token automatically.

## Links
Relates to: #5348

### CODE REVIEW

File: scripts/actions/utils/vendor-request.js [hunk 1]
```diff
@@ -68,25 +69,35 @@ const makeRequest = async (url, options, nthTry = 1) => {
...
+    console.log('grabbing access token');
+    const { accessToken } = await makeRequest(url, options);
 
-  const { accessToken } = await makeRequest(url, options);
+    console.log('setting cached token');
+    cache.set('access_token', accessToken, 60 * 4);
 
-  return accessToken;
+    return accessToken;
+  }
```
Comment (josephgregoryii) [inline scripts/actions/utils/vendor-request.js:100]: Very clean solution

**Review by josephgregoryii — APPROVED** [review]

**Comment by josephgregoryii** [comment]
Looks like there are some tests that are not passing. I think it is due to the length of arguments now that some of the function signatures have changed in some files.

**Review by josephgregoryii — DISMISSED** [review]
Looks like there are some tests that are not passing. I think it is due to the length of arguments now that some of the function signatures have changed in some files.

File: scripts/actions/check-job-progress.js [hunk 2]
```diff
@@ -74,7 +72,6 @@ const getBatchStatus = (accessToken) => async ({ batchUid, jobId }) => {
     const jobData = await vendorRequest({
       method: 'GET',
       endpoint: `/jobs-api/v3/projects/${PROJECT_ID}/jobs/${translationJobUid}`,
```
Comment (rudouglas) [inline scripts/actions/check-job-progress.js:74]: Do we not need any authorization for these api calls?

File: scripts/actions/check-job-progress.js [hunk 3]
```diff
@@ -116,10 +113,7 @@ const main = async () => {
     });
 
     // get the status for all of the batch translation jobs
-    const accessToken = await getAccessToken();
-    const batchStatuses = await Promise.all(
-      batchUids.map(getBatchStatus(accessToken))
-    );
+    const batchStatuses = await Promise.all(batchUids.map(getBatchStatus));
```
Comment (rudouglas) [inline scripts/actions/check-job-progress.js:116]:
```
    const batchStatuses = test()
```

File: scripts/actions/check-job-progress.js (hunk 2 above)
Comment (nr-kkenney) [reply scripts/actions/check-job-progress.js:74]: we do. but now that specific concern is handled entirely inside of vendorRequest.

Comment (rudouglas) [reply scripts/actions/check-job-progress.js:74]: ahhh right ye that makes sense to me noooow

**Review by rudouglas — APPROVED** [review]

**Pull request merged**